### ISSUE TITLE
A rest parameter must be of an array type.

### ISSUE THREAD

**Issue by TheColorRed** [Microsoft/TypeScript#23440]
When running a rest type with multiple array types an error is thrown. Using `(xx | yy)[]` works however that is not always desired, due to the fact that it allows for mixing types in one array, and not two different types of arrays.

**TypeScript Version:**  2.8.1

**Code**

```ts
function dist(x1: number, y1: number, x2: number, y2: number): number
function dist(point1: point, point2: point): number
function dist(...args: number[] | point[]): number {
  // function body
}
```

**Expected behavior:**

`...args` should not throw an error

**Actual behavior:**

The following error is thrown

> A rest parameter must be of an array type.

**Playground Link:** 

[Playground](http://www.typescriptlang.org/play/#src=interface%20point%20%7Bx%3A%20number%2C%20y%3A%20number%7D%0D%0A%0D%0Afunction%20dist(x1%3A%20number%2C%20y1%3A%20number%2C%20x2%3A%20number%2C%20y2%3A%20number)%3A%20number%0D%0Afunction%20dist(point1%3A%20point%2C%20point2%3A%20point)%3A%20number%0D%0Afunction%20dist(...args%3A%20number%5B%5D%20%7C%20point%5B%5D)%3A%20number%20%7B%0D%0A%20%20%2F%2F%20Run%20the%20code%0D%0A%7D)
